src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -27,4 +27,41 @@ public Integer getLength() {
     public String getColor() {
         return color;
     }
+
+    @Override
+    public boolean equals(Object object) {
+        if (this == object) {
+            return true;
+        }
+
+        if (object == null) {
+            return false;
+        }
+
+        if (getClass() != object.getClass()) {
+            return false;
+        }
+
+        Rectangle rectangle = (Rectangle) object;
+        if (width == rectangle.width
+                || (width != null && width.equals(rectangle.width))
+                && (length == rectangle.length)
+                || (length != null && length.equals(rectangle.length))
+                && (color == rectangle.color)
+                || (color != null && color.equals(rectangle.color))) {
+            return true;
```

Comment:
```suggestion
return (width == rectangle.width 
                || width != null && width.equals(rectangle.width))
                && (length == rectangle.length 
                || length != null && length.equals(rectangle.length))
                && (color == rectangle.color 
                || color != null && color.equals(rectangle.color));
```

Reply:
The logical-AND operator ( && ) has higher precedence than the logical-OR operator ( || )
add parenthesis to this statement to ensure the flow is correct

Reply:
done